### PR TITLE
Add some PHPDoc return types and fixes

### PR DESCRIPTION
**Pull request by franmomu**
Kind of follow-up of https://github.com/stripe/stripe-php/pull/1230

It also fixes some PHPDoc (found by PHPStan raising the current level)

r? @pakrym-stripe

### CODE REVIEW

File: lib/Invoice.php [hunk 1]
```diff
@@ -154,7 +154,7 @@ class Invoice extends ApiResource
      * @param null|array $params
      * @param null|array|string $opts
      *
-     * @throws StripeExceptionApiErrorException if the request fails
+     * @throws Stripe\Exception\ApiErrorException if the request fails
```
Comment (szepeviktor) [inline lib/Invoice.php:157]:
```suggestion
     * @throws \Stripe\Exception\ApiErrorException if the request fails
```

**Comment by szepeviktor** [comment]
🍏 CI says _okay_
https://github.com/szepeviktor/stripe-php/actions/runs/2369574313

**Review by pscheit — APPROVED** [review]

**Comment by pscheit** [comment]
@franmomu thanks for that, can you check the conflicts?

**Comment by pscheit** [comment]
now we just need someone to push the CI :) @pakrym-stripe maybe? :)

**Comment by pakrym-stripe** [comment]
Thank you, friends!

**Pull request merged**